src/main/java/duke/utility/Ui.java

```diff
@@ -0,0 +1,43 @@
+package duke.utility;
+
+import java.util.Scanner;
+
+public class Ui {
+
+    public Ui() {
+    }
+
+    /**
+     * Starts listening for commands from the user.
+     * @param tasks the {@link duke.utility.TaskList} this Ui is linked to.
+     * @param storage the {@link duke.utility.Storage} this Ui is linked to.
+     */
+    public void startListening(TaskList tasks, Storage storage) {
+        Parser parser = new Parser(tasks, storage);
+        Scanner sc = new Scanner(System.in);
+        String welcomeMessage = "Hello I'm Duke!\nWhat can I do for you?";
+        this.printMessage(welcomeMessage);
+
+        boolean shouldContinue = true;
```

Comment:
Maybe this boolean could be made more specific by naming it 'shouldContinueListening' instead of shouldContinue, it may be easier for readers to know what is being continued.